centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1442,26 +1455,33 @@ objDataToMoney(int data_type, void* data_ptr, pMoneyType m)
 		    ptr++;
 		    }
 		intval = 0;
-		m->FractionPart = 0;
-		m->WholePart = 0;
-		intval = strtoul(ptr, &endptr, 10);
-		if (intval > 0x7FFFFFFFUL)
+		m->MoneyValue = 0;
+		intval = strtoll(ptr, &endptr, 10);
+		if (intval > 0x7FFFFFFFFFFFFFFFLL)
 		    return -1;
 		if ((endptr - ptr) != strspn(ptr, "0123456789"))
 		    return -1;
 		if (is_neg)
-		    m->WholePart = -intval;
+		    m->MoneyValue = -intval*10000;
 		else
-		    m->WholePart = intval;
+		    m->MoneyValue = intval*10000;
 		if (*endptr == (intl_format?',':'.'))
 		    {
-		    intval = strtoul(endptr+1, &endptr2, 10);
+		    intval = strtoll(endptr+1, &endptr2, 10);
 		    scale = endptr2 - (endptr+1);
 		    if (scale != strspn(endptr+1, "0123456789"))
 			return -1;
 		    while(scale < 4) { scale++; intval *= 10; }
 		    while(scale > 4) { scale--; intval /= 10; }
-		    m->FractionPart = intval;
+		    if (is_neg)
+                {
```

Comment:
maybe double check whether you're using the same spaces/tabs as the surrounding code?

Reply:
and whether CLion is doing autoformatting